sys/parsoid_proxy.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+/*
+ * Simple proxy to route requests to the client-requested
+ * Parsoid variant (JS or PHP) during the transistion
+ * persiod. Parsoid/JS is being phased out and replaced by
+ * Parsoid/PHP.
+ */
+
+const HyperSwitch = require('hyperswitch');
+const URI = HyperSwitch.URI;
+const HTTPError = HyperSwitch.HTTPError;
+const mwUtil = require('../lib/mwUtil');
+
+module.exports = (options) => {
+    options = options || {};
+    const phpHost = options.parsoidPhpHost;
+    const VARIANT_HDR_NAME = 'x-parsoid-variant';
+    return {
+        spec: {
+            paths: {
+                '/{+path}': {
+                    'x-route-filters': options.block_external_reqs ?
+                        [{
+                            type: 'default',
+                            name: 'header_match',
+                            options: {
+                                whitelist: {
+                                    'x-client-ip': ['/^(?:::1)|(?:::ffff:)?(?:10|127)\\./']
+                                }
+                            }
+                        }] : [],
+                    all: {
+                        operationId: 'proxy',
+                        'x-monitor': false,
+                        'x-hidden': true
+                    }
+                }
+            }
+        },
+        operations: {
+            proxy: (hyper, req) => {
+                const rootReqHeaders = hyper._rootReq.headers;
+                if (!rootReqHeaders[VARIANT_HDR_NAME]) {
+                    rootReqHeaders[VARIANT_HDR_NAME] = 'JS';
+                }
+                const isPhpVariant = /PHP/i.test(rootReqHeaders[VARIANT_HDR_NAME]);
+                if (isPhpVariant && !phpHost) {
```

Comment:
This is superfluous here because RB will fail to load if both modules don't get their respective host option.